benchmarks/definitions/nashpobench.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+from pathlib import Path
+import numpy as np
+from random import shuffle
+
+from sklearn.ensemble import RandomForestRegressor
+
+from syne_tune.search_space import choice, randint, uniform, loguniform
+
+from blackbox_repository.conversion_scripts.scripts.fcnet_import import \
+    METRIC_ELAPSED_TIME, METRIC_VALID_LOSS, RESOURCE_ATTR, BLACKBOX_NAME
+
+
+class SubsamplingSurrogate(object):
+
+    def __init__(self, ratio=0.01):
+        self.ratio = ratio
+
+    def fit(self, X, y):
+        n = int(X.shape[0] * self.ratio)
+        idx = [i for i in range(X.shape[0])]
+
+        shuffle(idx)
+        idx = idx[:n]
+        X_train = X[idx]
+        y_train = np.array(y)[idx]
+
+        self.model = RandomForestRegressor()
+        self.model.fit(X_train, y_train)
+        return self
+
+    def predict(self, X_test):
+        return self.model.predict(X_test)
+
+
+_config_space = {
+    "hp_activation_fn_1": choice(["tanh", "relu"]),
+    "hp_activation_fn_2": choice(["tanh", "relu"]),
+    "hp_batch_size": randint(8, 64),
+    "hp_dropout_1": uniform(0.0, 0.6),
+    "hp_dropout_2": uniform(0.0, 0.6),
+    "hp_init_lr": loguniform(0.0005, 0.1),
+    'hp_lr_schedule': choice(["cosine", "const"]),
+    'hp_n_units_1': randint(16, 512),
+    'hp_n_units_2': randint(16, 512),
+}
+
+
+def nashpobench_default_params(params=None):
+    dont_sleep = str(
+        params is not None and params.get('backend') == 'simulated')
+    return {
+        'max_resource_level': 100,
+        'grace_period': 1,
+        'reduction_factor': 3,
+        'instance_type': 'ml.m5.large',
+        'num_workers': 4,
+        'framework': 'PyTorch',
+        'framework_version': '1.6',
+        'dataset_name': 'protein_structure',
+        'dont_sleep': dont_sleep,
+        'cost_model_type': 'linear',
```

Comment:
This probably does not apply. The cost model for NASBench201 is very specific. We can spend extra work to define a fitting cost model for this benchmark here as well, but I think it is not there yet.

Reply:
ok can I just pass `None` ?

Reply:
Just remove the line.